google_compute_engine/metadata_scripts/script_retriever.py

```diff
@@ -50,29 +55,35 @@ def _DownloadGsUrl(self, url, dest_dir):
     Returns:
       string, the path to the file storing the metadata script.
     """
-    try:
-      subprocess.check_call(
-          ['which', 'gsutil'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    except subprocess.CalledProcessError:
-      self.logger.warning(
-          'gsutil is not installed, cannot download items from Google Storage.')
-      return None
-
     dest_file = tempfile.NamedTemporaryFile(dir=dest_dir, delete=False)
     dest_file.close()
     dest = dest_file.name
 
-    self.logger.info('Downloading url from %s to %s using gsutil.', url, dest)
+    self.logger.info(
+        'Downloading url from %s to %s using authentication token.',
+        url, dest)
     try:
-      subprocess.check_call(['gsutil', 'cp', url, dest])
+      if not self._token:
+        tok_metadata = 'instance/service-accounts/default/token'
+        response = self.watcher.GetMetadata(tok_metadata, recursive=False)
+        self._token = '%s %s' % (
+            response[u'token_type'], response[u'access_token']
+        )
+
+      request = urlrequest.Request(url)
+      request.add_unredirected_header('Metadata-Flavor', 'Google')
+      request.add_unredirected_header('Authorization', self._token)
+      content = urlrequest.urlopen(request).read()
+
+      with open(dest, 'w') as f:
+        f.write(content)
+
       return dest
-    except subprocess.CalledProcessError as e:
-      self.logger.warning(
-          'Could not download %s using gsutil. %s.', url, str(e))
     except Exception as e:
```

Comment:
This try except block seems fairly large. You should scope the block to areas that throw exceptions (potentially have two), and then handle the exceptions you expect. I don't think you should need to catch a generic Exception.